protocols/floodsub/src/protocol.rs

```diff
@@ -58,8 +111,18 @@ where
 
     fn upgrade_inbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
         Box::pin(async move {
-            let packet = upgrade::read_length_prefixed(&mut socket, 2048).await?;
-            let rpc = rpc_proto::Rpc::decode(&packet[..]).map_err(DecodeError)?;
+            // From Thomas:
+            // What you probably want to do here is use a Framed to wrap
+            // the incoming socket and immediately read a message from it
+            // using its Stream::next implementation.
+            //
+            // Framed is an abstraction on top of a codec to
+            // read and write from a socket in entire messages instead of
+            // raw bytes!
+
+            let mut framed =
+                Framed::<TSocket, FloodsubProtocol>::new(socket, FloodsubProtocol::new());
```

Comment:
```suggestion
                Framed::<TSocket, prost_codec<rpc_proto::Rpc>>::new(socket, prost_codec::Codec::new(MAX_MESSAGE_LEN_BYTES));
```

We can probably do this actually and delete:

- The `Encoder` / `Decoder` impls
- Manual `Clone`, `Debug` and `Default` impls